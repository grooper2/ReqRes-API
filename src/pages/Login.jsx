import React, {useState} from 'react';
import { useHistory } from "react-router-dom";
import regresRepository from '../repositories/resregRepositories';
import "../App.css";

const Login = () =>{
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const history = useHistory();

    const onChangeEmail = async (e) => {
        setEmail(e.target.value);
    }

    const onChangePassword = async (e) => {
        setPassword(e.target.value);
    };

    const onSubmit = async (e) =>{
        e.preventDefault();
        submitForm();
    };

    const submitForm = async () => {
        await regresRepository.loginRequest({
            email: email,
            password: password,
        });
        history.push("/home");
    };

    
        return (
            <div className="container center">
                <div className="card">  
                    <h2>REQRES-API</h2>
                    <hr />
                    <h3>Welcome back to REQRES-API</h3>
                    <div className="registerFormTitle">
                        <p>Login</p>
                    </div>
                    <form onSubmit={onSubmit}>
                        <div className="form-field center">
                            <label className="input">
                                <input className="input__field" type="text" placeholder=" " onChange={onChangeEmail} />
                                <span className="input__label">Email</span>
                            </label>
                        </div>
                        <div className="form-field center">
                            <label className="input">
                                <input className="input__field" type="password" placeholder=" " onChange={onChangePassword} />
                                <span className="input__label">Password</span>
                            </label>
                        </div>
                        <div className="form-field center">
                            <input type="submit" className="register-btn" value="Login"/>
                            <p>Don't have an account? Click <a className="clickHere"onClick={()=>{history.push("/")}}>Here</a> to Register</p>
                        </div>
                    </form>
                </div>
            </div>
        )
    
}

export default Login;
