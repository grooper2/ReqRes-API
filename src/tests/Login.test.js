import React from "react";
import { render, waitFor, fireEvent } from "@testing-library/react";
import Login from "../pages/Login";

describe("<Login />", () => {
  it("renders and displays properly", async () => {
    const { container } = render(<Login />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});