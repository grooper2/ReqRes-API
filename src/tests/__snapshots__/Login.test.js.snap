// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders and displays properly 1`] = `
<div>
  <div
    class="container center"
  >
    <div
      class="card"
    >
      <h2>
        REQRES-API
      </h2>
      <hr />
      <h3>
        Welcome back to REQRES-API
      </h3>
      <div
        class="registerFormTitle"
      >
        <p>
          Login
        </p>
      </div>
      <form>
        <div
          class="form-field center"
        >
          <label
            class="input"
          >
            <input
              class="input__field"
              placeholder=" "
              type="text"
            />
            <span
              class="input__label"
            >
              Email
            </span>
          </label>
        </div>
        <div
          class="form-field center"
        >
          <label
            class="input"
          >
            <input
              class="input__field"
              placeholder=" "
              type="password"
            />
            <span
              class="input__label"
            >
              Password
            </span>
          </label>
        </div>
        <div
          class="form-field center"
        >
          <input
            class="register-btn"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
